lib/tasks/data/export_team.rake

```diff
@@ -1,320 +1,459 @@
-          else
-            raise "Missing query to copy #{table}"
-          end
-        rescue Exception => e
-          raise "Error dumping table #{table}: #{e.message}"
-        end
-        begin
-          ActiveRecord::Base.connection.execute("DROP TABLE IF EXISTS users_outside_#{@team.id};") if table == 'users'
-        rescue Exception => e
-        end
+      team_id = team.id
+      filename = "check-#{team.slug}.sqlite3"
+      begin
+        File::unlink(filename)
+        puts "Deleted existing #{filename}"
+      rescue Errno::ENOENT
+        # Common case
       end
-      export_zip
-      FileUtils.remove_dir(@tmp_folder_path, true) if File.exist?(@tmp_folder_path)
-    end
+      begin
+        File::unlink("#{filename}.lz4")
+        puts "Deleted existing #{filename}.lz4"
+      rescue Errno::ENOENT
+        # Common case
+      end
+      SQLite3::Database.new(':memory:') do |db|
+        for klass in [
+            TableStrategies::Account,
+            TableStrategies::AccountSource,
+            TableStrategies::Annotation,
+            TableStrategies::Assignment,
+            TableStrategies::Bounce,
+            TableStrategies::Contact,
+            TableStrategies::DynamicAnnotationAnnotationType,
+            TableStrategies::DynamicAnnotationField,
+            TableStrategies::DynamicAnnotationFieldInstance,
+            TableStrategies::DynamicAnnotationFieldType,
+            TableStrategies::LoginActivity,
+            TableStrategies::Media,
+            TableStrategies::Project,
+            TableStrategies::ProjectMedia,
+            TableStrategies::ProjectMediaProject,
+            TableStrategies::Relationship,
+            TableStrategies::Source,
+            TableStrategies::TagText,
+            TableStrategies::Team,
+            TableStrategies::TeamTask,
+            TableStrategies::TeamUser,
+            TableStrategies::User,
+            #TableStrategies::Version,
+        ]
+          dumper = klass.new(team_id)
+          puts "Dumping #{dumper.table_name}..."
+          dumper.copy_to_sqlite3_database(db)
+        end
 
-    def table(name)
-      if name.match(/annotations_(.*)/)
-        'annotations'
-      elsif name.match(/versions_(.*)/)
-        'versions'
-      else
-        name
+        puts "Writing #{filename}..."
+        # We _could_ use SQLite3::Database.new(filename) directly. But testing
+        # this on 2020-07-23:
+        #
+        # time docker-compose exec api bundle exec rake check:data:export_team[boom-factcheck]
+        #
+        # gave these results:
+        #
+        # DIRECT TO FILE: 30.8s, 349MB (63MB gzipped)
+        # MEMORY AND VACUUM: 26.5s, 327MB (54MB gzipped)
+        #
+        # So VACUUM INTO is 14% faster (or much more, if we ignore Rails init)
+        # and 6% more space-efficient (or 14%, after compression)
+        db.execute("VACUUM INTO '#{filename}'")
       end
+
+      # We compress for uploading to Workbench with LZ4 compression.
```

Comment:
No need to mention Workbench here - my understanding is that .sqllite.lz4 is fairly agnostic, right?

Reply:
I think it's a rare combination. Google finds plenty of filenames ending in ".sqlite3.gz" on the Web. It finds zero matches for ".sqlite3.lz4".

We'd be using `.sqlite3` if size didn't matter. (Size does matter, because Workbench limits the total size of uploaded files per step.) We'd be using `.sqlite3.gz` if the decompression only happened once. (Decompression happens every run, because Workbench wipes all data after every execution.)

If I were reading somebody else's code and it didn't explain that we're uploading to Workbench, I would think ".sqlite.lz4" is outlandish. I